src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ProjectDebuggerProvider.cs

```diff
@@ -0,0 +1,338 @@
+    [AppliesTo(ProjectCapability.LaunchProfiles)]
+    internal class ProjectDebuggerProvider : DebugLaunchProviderBase, IDeployedProjectItemMappingProvider
+    {
+
+        /// <summary>
+        /// Constructors. Unit test one is 2nd
+        /// </summary>
+        [ImportingConstructor]
+        public ProjectDebuggerProvider(ConfiguredProject configuredProject, ILaunchSettingsProvider launchSettingsProvider)
+            : base(configuredProject)
+        {
+            LaunchSettingsProvider = launchSettingsProvider;
+
+            // We want it sorted so that higher numbers come first (is the default for these collections but explicitly expressed here)
+            ProfileLaunchTargetsProviders = new OrderPrecedenceImportCollection<IDebugProfileLaunchTargetsProvider>(ImportOrderPrecedenceComparer.PreferenceOrder.PreferredComesFirst, 
+                                                                                                                    configuredProject.UnconfiguredProject);
+        }
+
+        public ProjectDebuggerProvider(ConfiguredProject configuredProject, ILaunchSettingsProvider launchSettingsProvider, 
+                                       OrderPrecedenceImportCollection<IDebugProfileLaunchTargetsProvider> providers)
+            : base(configuredProject)
+        {
+            ProfileLaunchTargetsProviders = providers;
+            LaunchSettingsProvider = launchSettingsProvider;
+        }
+
+        /// <summary>
+        /// Import the LaunchTargetProviders which know how to run profiles
+        /// </summary>
+        [ImportMany]
+        private OrderPrecedenceImportCollection<IDebugProfileLaunchTargetsProvider> ProfileLaunchTargetsProviders { get; set; }
+
+        private ILaunchSettingsProvider LaunchSettingsProvider { get; }
+
+        // Tracks the last launched provider so we can forward calls to IDeployedProjectItemMappingProvider
+        public IDebugProfileLaunchTargetsProvider LastLaunchProvider { get; private set; }
+
+        /// <summary>
+        /// Called by CPS to determine whether we can launch
+        /// </summary>
+        public override Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
+        {
+            return TplExtensions.TrueTask;
+        }
+
+        /// <summary>
+        /// Helper returns the correct debugger engine based on the targeted framework
+        /// </summary>
+        public static Guid GetDebugEngineForFramework(string targetFramework)
+        {
+            // The engine depends on the framework
+            if (IsDotNetCoreFramework(targetFramework))
+            {
+                return  DebuggerEngines.ManagedCoreEngine;
+            }
+            else
+            {
+                return DebuggerEngines.ManagedOnlyEngine;
+            }
+        }
+
+        /// <summary>
+        /// This is a placeholder
+        /// </summary>
+        public static bool IsDotNetCoreFramework( string targetFramework)
+        {
```

Comment:
This is the only place they are used and this code is only temporary until issue #443 is address